js/rcMenuFuncs.js

```diff
@@ -16,131 +16,134 @@
-  textarea.classList.add(target.id);
-  textarea.rows = '1';
-  setRows(textarea);
-  target.appendChild(textarea);
-
-  textarea.addEventListener('keydown', (e) => {
-    if (e.keyCode === 13 && !e.shiftKey) {
-      if (textarea.value == text) return editDOM(target, textarea, text);
-      let newText = textarea.value;
-      newText = newText.replace(/(<a?:)(!)?(.+?:[0-9]+?>)/gm, (a, b, c, d) => {
-        if (c != '!') {
-          return `${b}!${d}`;
+    // Safe check so there is only one DOM to edit messages
+    checkEditDoms();
+
+    // Find and delete the message text <p> element
+    let textElement = target.querySelector('.messageText');
+
+    // Get the text in the message
+    let elementText = textElement.innerHTML;
+    let msg = selectedChan.messages.cache.get(target.id);
+    let text = msg.cleanContent;
+
+    target.removeChild(textElement);
+
+    let textarea = document.createElement('textarea');
+    textarea.value = text;
+    textarea.classList.add('editTextarea');
+    textarea.classList.add('messageBoxText');
+    textarea.classList.add(target.id);
+    textarea.rows = '1';
+    setRows(textarea);
+    target.appendChild(textarea);
+
+    textarea.addEventListener('keydown', (e) => {
+        if (e.keyCode === 13 && !e.shiftKey) {
+            if (textarea.value == text) return editDOM(target, textarea, text);
+            let newText = textarea.value;
+            newText = newText.replace(
+                /(<a?:)(!)?(.+?:[0-9]+?>)/gm,
+                (a, b, c, d) => {
+                    if (c != '!') {
+                        return `${b}!${d}`;
+                    }
+                    return a;
+                }
+            );
+
+            newText = parseSend(newText);
+
+            msg.edit(newText).catch((e) => {
+                command('Message failed to send\nError: ' + e.message);
+            });
+
+            editDOM(target, textarea, elementText);
         }
-        return a;
-      });
-
-      newText = parseSend(newText);
-
-      msg.edit(newText).catch((e) => {
-        command('Message failed to send\nError: ' + e.message);
-      });
+    });
 
-      editDOM(target, textarea, elementText);
-    }
-  });
-
-  textarea.addEventListener('input', (e) => setRows(textarea));
+    textarea.addEventListener('input', (e) => setRows(textarea));
 }
 
 function setRows(textarea) {
-  let rows = textarea.value.split('\n').length;
-  if (rows > 6) rows = 6;
-  if (rows == 0) rows++;
-  textarea.rows = rows;
+    let rows = textarea.value.split('\n').length;
+    if (rows > 6) rows = 6;
+    if (rows == 0) rows++;
+    textarea.rows = rows;
 }
 
 function pinMsg(id, pin) {
-  try {
-    pin
-      ? selectedChan.messages.cache.get(id).pin()
-      : selectedChan.messages.cache.get(id).unpin();
-  } catch (e) {
-    console.log(e);
-  }
+    try {
+        pin
+            ? selectedChan.messages.cache.get(id).pin()
+            : selectedChan.messages.cache.get(id).unpin();
```

Comment:
What the f--- kinda ternary operation did daniel add here...?? May as well just use a normal if statement at this point.

Reply:
That checks if `pin` exists, then if it does, pins the message, otherwise, does not.